src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -548,42 +546,135 @@ private void SetDirty(TimeSpan? maxDelay)
             return applicationData;
         }
 
-        private async Task InitializeAsync(bool ignoreFailures, IEnumerable<ConfigurationClient> availableClients, CancellationToken cancellationToken = default)
+        private async Task LoadAsync(bool ignoreFailures, CancellationToken cancellationToken)
         {
-            Dictionary<string, ConfigurationSetting> data = null;
-            Dictionary<KeyValueIdentifier, ConfigurationSetting> watchedSettings = null;
-            
+            var startupStopwatch = Stopwatch.StartNew();
+
+            int postFixedWindowAttempts = 0;
+
+            var startupExceptions = new List<Exception>();
+
+            IEnumerable<ConfigurationClient> clients = _configClientManager.GetAllClients();
```

Comment:
Move this into the loop so you get new clients when replica auto discovery is enabled.